control-plane/connect-inject/endpoints_controller.go

```diff
@@ -1072,3 +1093,12 @@ func consulTags(pod corev1.Pod) []string {
 
 	return interpolatedTags
 }
+
+func getMultiPortIdx(pod corev1.Pod, serviceEndpoints corev1.Endpoints) int {
+	for i, name := range strings.Split(pod.Annotations[annotationService], ",") {
+		if name == getServiceName(pod, serviceEndpoints) {
+			return i
+		}
+	}
+	return -1
```

Comment:
nit: I think we could also return an error here (it feels a bit more idiomatic to go) but definitely not a blocker!